app/code/Magento/Vault/view/frontend/templates/cards_list.phtml

```diff
@@ -33,3 +33,11 @@ $tokens = $block->getPaymentTokens();
 <?php elseif (!$block->isExistsCustomerTokens()): ?>
     <div class="message info empty"><span><?php /* @noEscape */ echo __('You have no stored payment methods.'); ?></span></div>
 <?php endif ?>
+
+<div class="actions-toolbar">
```

Comment:
This container can't be located after stored cards because the block with stored accounts also should be rendered. And it will look at least confusing.
I propose to create an own block for the `Back` button and add it via `vault_cards_listaction.xml` layout.